src/git.md

```diff
@@ -0,0 +1,191 @@
+to rust-lang/rust since then are in conflict with the changes you've made.
+
+When this happens, you need to resolve the conflicts before your changes can be
+merged. First, get a local copy of the conflicting changes: Checkout your local
+master branch with `git checkout master`, then `git pull rust master` to
+update it with the most recent changes.
+
+### Rebasing
+
+You're now ready to start the rebasing process. Check out the branch with your
+changes and execute `git rebase master`.
+
+When you rebase a branch on master, all the changes on your branch are
+reapplied to the most recent version of master. In other words, git tries to
+pretend that the changes you made to the old version of master were instead
+made to the new version of master. During this process, you should expect to
+encounter at least one "rebase conflict." This happens when git's attempt to
+reapply the changes fails because your changes conflicted with other changes
+that have been made. You can tell that this happened because you'll see
+lines in the output that look like
+
+```
+CONFLICT (content): Merge conflict in file.rs
+```
+
+When you open these files, you'll see sections of the form
+
+```
+<<<<<<< HEAD
+Original code
+=======
+Your code
+>>>>>>> 8fbf656... Commit fixes 12345
+```
+
+This represents the lines in the file that git could not figure out how to
+rebase. The section between `<<<<<<< HEAD` and `=======` has the code from
+master, while the other side has your version of the code. You'll need to
+decide how to deal with the conflict. You may want to keep your changes,
+keep the changes on master, or combine the two.
+ 
+Generally, resolving the conflict consists of two steps: First, fix the
+particular conflict. Edit the file to make the changes you want and remove the
+`<<<<<<<`, `=======` and `>>>>>>>` lines in the process. Second, check the
+surrounding code. If there was a conflict, its likely there are some logical
+errors lying around too! It's a good idea to run `x.py check` here to make sure
+there are no glaring errors.
+
+Once you're all done fixing the conflicts, you need to stage the files that had
+conflicts in them via `git add`. Afterwards, run `git rebase --continue` to let
+git know that you've resolved the conflicts and it should finish the rebase.
+Once the rebase has succeeded, you'll want to update the associated branch on
+your fork with `git push --force`.
+
+Note that `git push` will not work properly and say something like this:
+
+```
+ ! [rejected]        issue-xxxxx -> issue-xxxxx (non-fast-forward)
+error: failed to push some refs to 'https://github.com/username/rust.git'
+hint: Updates were rejected because the tip of your current branch is behind
+hint: its remote counterpart. Integrate the remote changes (e.g.
+hint: 'git pull ...') before pushing again.
+hint: See the 'Note about fast-forwards' in 'git push --help' for details.
+```
+
+The advice this gives is incorrect! Because of the "no-merge" policy, running
```

Comment:
You'll probably want to apply this manually, but perhaps change to something like this?

```suggestion
The advice this gives is incorrect! Because of Rust's ["no-merge" policy](#no-merge-policy), running
```